core/src/main/scala/scala/pickling/pickler/Any.scala

```diff
@@ -12,32 +12,44 @@ import scala.reflect.runtime.currentMirror
 object AnyPicklerUnpickler extends AbstractPicklerUnpickler[Any]
     with AutoRegister[Any] {
 
+  final val nullPickler = Defaults.nullPickler.asInstanceOf[Pickler[Any]]
+
   override def tag: FastTypeTag[Any] = FastTypeTag.Any
 
   /** Pickle [[Any]] by getting its class at runtime and looking
     * up the correct [[Pickler]] for that class if available or
     * generate it.
     *
     * Don't use [[AnyPicklerUnpickler]] for pickling classes with generic
-    * types. Otherwise, it will fail because of the type erasure
-    * and the lookup will replace the unknown type by [[Any]].
+    * types. Otherwise, it will fail because of the type erasure.
+    * The lookup will replace the unknown type parameters by [[Any]].
     */
   override def pickle(picklee: Any, builder: PBuilder): Unit = {
-    val clazz = picklee.getClass
-    val classLoader = this.getClass.getClassLoader
-    GRL.lock()
-    val tag = try FastTypeTag.makeRaw(clazz)
-    finally GRL.unlock()
-    val p = currentRuntime.picklers.genPickler(classLoader, clazz, tag)
-    p.asInstanceOf[Pickler[Any]].pickle(picklee, builder)
+
+    // Use nullPickler if null, get pickler otherwise
+    val pickler = if (picklee == null) nullPickler else {
```

Comment:
Yes, probably it would be better to remove the null checks in higher layers of the interface and just reuse this :smiley: